src/strategies/shared.rs

```diff
@@ -49,7 +52,10 @@ pub trait UniswapXStrategy {
         let fill_contract =
             UniversalRouterExecutor::new(Address::from_str(executor_address)?, client.clone());
 
-        let token_in = Address::from_str(&request.token_in)?;
+        // Normalize ERC20ETH to native ETH (zero address) since we'll receive native ETH during callback
+        let token_in = normalize_erc20eth_to_native_address(
+            Address::from_str(&request.token_in)?
```

Comment:
I slightly prefer something like `request.token_in.parse::<Address>()?` or simply `request.token_in.parse()?` but entirely up to you